File: src/core/candidate/dto/create-candidate.dto.ts
import { IsString, IsNotEmpty, IsMongoId } from 'class-validator';

export class CreateCandidateDto {
  @IsString({ message: 'Digite um nome válido' })
  @IsNotEmpty({ message: 'Digite um nome válido' })
  public name: string;

  @IsMongoId({ message: 'Eleição inválida' })
  public election: string;
}

File: src/main/java/dev/joaopdias/vox/core/candidate/dto/CreateCandidateDto.java
package dev.joaopdias.vox.core.candidate.dto;

import jakarta.validation.constraints.NotNull;

import java.util.UUID;

public record CreateCandidateDto(
        @NotNull(message = "Digite um nome válido")
        String name,

        @NotNull(message = "Selecione uma eleição válida")
        UUID electionId
) {
}
